news-bundle/src/Resources/contao/templates/news/news_latest.html5

```diff
@@ -20,3 +20,11 @@
   <?php endif; ?>
 
 </div>
+
+<?php
+
+// This template is used as a news list template by default, so we only add
+// JSON-LD data in case this is a news article without a reader
+if (!$this->hasReader) {
```

Comment:
Why did you not use `if (false === $this->hasText) {`? Then we would not have to add a new variable.

Reply:
Because for external links, `hasText` is `true` too. But for us, only `$this->source === 'default'` is relevant.

Reply:
Well, `$this->source` does not exist?

Reply:
It should exist:

```php
		$objTemplate = new FrontendTemplate($this->news_template ?: 'news_latest');
		$objTemplate->setData($objArticle->row());
```

Reply:
Ah true. Well...yeah. We could. I don't really mind. I find `hasReader` tells me more than "the default source" :)

Reply:
What about my `$this->isExternal` suggestion above? I find `hasReader` not very clear.

Reply:
But `isExternal` is not true? We want to know which ones have a reader page.